solidity/ecdsa/test/WalletRegistry.Authorization.test.ts

```diff
@@ -0,0 +1,1854 @@
+          await staking
+            .connect(authorizer)
+            ["requestAuthorizationDecrease(address,address,uint96)"](
+              stakingProvider.address,
+              walletRegistry.address,
+              deauthorizingBy
+            )
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        context("when sortition pool was not yet updated", () => {
+          it("should return false", async () => {
+            expect(await walletRegistry.isOperatorUpToDate(operator.address)).to
+              .be.false
+          })
+        })
+
+        context("when the sortition pool was updated", () => {
+          it("should return true", async () => {
+            await walletRegistry.updateOperatorStatus(operator.address)
+            expect(await walletRegistry.isOperatorUpToDate(operator.address)).to
+              .be.true
+          })
+        })
+      })
+
+      context("when operator was slashed when the pool was locked", () => {
+        before(async () => {
+          await createSnapshot()
+
+          // Increase authorization to the maximum possible value and update
+          // sortition pool. This way, any slashing from `slasher` application
+          // will affect authorized stake amount for WalletRegistry.
+          const authorized = await staking.authorizedStake(
+            stakingProvider.address,
+            walletRegistry.address
+          )
+          const increaseBy = stakedAmount.sub(authorized)
+          await staking
+            .connect(authorizer)
+            .increaseAuthorization(
+              stakingProvider.address,
+              walletRegistry.address,
+              increaseBy
+            )
+          await walletRegistry.updateOperatorStatus(operator.address)
+
+          // lock the pool for DKG
+          await walletRegistry.connect(walletOwner.wallet).requestNewWallet()
+
+          // and slash!
+          await staking
+            .connect(slasher.wallet)
+            .slash(to1e18(100), [stakingProvider.address])
+          await staking.connect(thirdParty).processSlashing(1)
+
+          // unlock the pool by stopping DKG
+          await mineBlocks(params.dkgSeedTimeout)
+          await walletRegistry.notifySeedTimeout()
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        context("when sortition pool was not yet updated", () => {
+          it("should return false", async () => {
+            expect(await walletRegistry.isOperatorUpToDate(operator.address)).to
+              .be.false
+          })
+        })
+
+        context("when the sortition pool was updated", () => {
+          it("should return true", async () => {
+            await walletRegistry.updateOperatorStatus(operator.address)
+            expect(await walletRegistry.isOperatorUpToDate(operator.address)).to
+              .be.true
+          })
+        })
+      })
+    })
+  })
+})
```

Comment:
Would it be worth adding unit tests for `eligibleStake` and `remainingAuthorizationDecreaseDelay` functions?